src/main.rs

```diff
@@ -34,6 +34,21 @@ fn main() {
 
     #[cfg(target_os = "macos")]
     {
+        // incase of app bundle, we can just pass --disowned option straight away to bypass this check
+        #[cfg(not(debug_assertions))]
+        if std::env::args().find(|f| f == "--disowned").is_none() {
```

Comment:
minor nitpick: to me using `!any()` rather than `find().is_none()` makes more sense as we wouldn't be depending on the option being none for the boolean value